pkg/dao/repository.go

```diff
@@ -34,10 +31,52 @@ func (r repositoryDaoImpl) Create(newRepo api.CreateRepository) error {
 		return result.Error
 	}
 	if err := db.DB.Create(&newRepoConfig).Error; err != nil {
-		if err.Error() == uniqueConstraintErrMsg {
-			return fmt.Errorf("repository with this URL already belongs to organization")
+		if isUniqueViolation(err) {
+			return &Error{BadValidation: true, Message: "Repository with this URL already belongs to organization "}
 		}
 		return err
 	}
 	return nil
 }
+
+func (r repositoryDaoImpl) Fetch(orgId string, uuid string) api.Repository {
+	repo := api.Repository{}
+	repo.FromRepositoryConfiguration(r.fetch(orgId, uuid))
+	return repo
+}
+
+func (r repositoryDaoImpl) fetch(orgId string, uuid string) models.RepositoryConfiguration {
+	found := models.RepositoryConfiguration{}
+	db.DB.First(&found, "UUID = ? AND ORG_ID = ?", uuid, orgId)
+	return found
+}
+
+func isUniqueViolation(err error) bool {
+	pgError, ok := err.(*pgconn.PgError)
+	if ok {
+		if pgError.Code == "23505" {
+			return true
+		}
+	}
+	return false
+}
+
+func (r repositoryDaoImpl) FullUpdate(orgId string, uuid string, repoParams api.CreateRepository) error {
+	repoConfig := r.fetch(orgId, uuid)
+	if repoConfig.UUID == "" {
+		return &Error{NotFound: true, Message: "Could not find Repository with uuid " + uuid}
+	}
+	ApiToModel(&repoParams, &repoConfig)
+	result := db.DB.Model(&repoConfig).Updates(repoConfig)
+	if result.Error != nil {
+		return &Error{Message: result.Error.Error(), BadValidation: isUniqueViolation(result.Error)}
+	}
+	return nil
+}
+
+func ApiToModel(apiRepo *api.CreateRepository, repoConfig *models.RepositoryConfiguration) {
+	repoConfig.Name = apiRepo.Name
+	repoConfig.Arch = apiRepo.DistributionArch
+	repoConfig.Version = apiRepo.DistributionVersion
+	repoConfig.URL = apiRepo.URL
```

Comment:
is leaving out AccountID and OrgID intentional?

Reply:
This is kinda on purpose.  You don't want the user to be able to pass in org id or account id in the request